src/engine-decorator-provider.js

```diff
@@ -0,0 +1,15 @@
+// @flow
+export default class EngineDecoratorProvider implements IEngineDecoratorProvider {
```

Comment:
Great @Yuvalke, but few things are missing (I just gave pseudo code example):
1. Add type annotations to methods and members (also need to mark members as private `_`).
2. Don't export default but named (caused us some issues in the past).
3. Export `EngineDecoratorProvider` from playkit (this where the engine decorator framework is defined).